src/shopify/utils/common_utils.bal

```diff
@@ -0,0 +1,391 @@
+    http:Client httpClient = store.getHttpClient();
+    http:Request request = new;
+    var result = httpClient->delete(path, request);
+    if (result is error) {
+        return createError("Could not retrive data from the Shopify server.", result);
+    }
+    http:Response response = <http:Response>result;
+    return checkResponse(response);
+}
+
+function getResponseForPostCall(Store store, string path, http:Request request) returns http:Response|Error {
+    http:Client httpClient = store.getHttpClient();
+    var result = httpClient->post(path, request);
+    if (result is error) {
+        return createError("Could not retrive data from the Shopify server.", result);
+    }
+    http:Response response = <http:Response>result;
+    return checkResponse(response);
+}
+
+function getResponseForPutCall(Store store, string path, http:Request request) returns http:Response|Error {
+    http:Client httpClient = store.getHttpClient();
+    var result = httpClient->put(path, request);
+    if (result is error) {
+        return createError("Could not retrive data from the Shopify server.", result);
+    }
+    http:Response response = <http:Response>result;
+    return checkResponse(response);
+}
+
+function buildQueryParamtersFromFilter(Filter filter) returns string|Error {
+    string queryParams = "";
+    foreach var [key, value] in filter.entries() {
+        if (key == LIMIT && filter?.'limit is int) {
+            int 'limit = <int>filter?.'limit;
+            if ('limit < 1 || 'limit > PAGE_MAX_LIMIT) {
+                return createError("The max limit must be a positive integer less than " + PAGE_MAX_LIMIT.toString()
+                    + " (inclusive)");
+            }
+            queryParams += "&" + LIMIT + "=" + 'limit.toString();
+        } else if (filter[key] is int || filter[key] is string) {
+            queryParams += "&" + convertToUnderscoreCase(key) + "=" + filter[key].toString();
+        } else if (filter[key] is int[] || filter[key] is string[]) {
+            queryParams += "&" + convertToUnderscoreCase(key) + "=" + buildCommaSeparatedListFromArray(<any[]>filter[key]);
+        } else if (filter[key] is DateFilter) {
+            DateFilter dateFilter = <DateFilter>filter[key];
+            string? before = getTimeStringTimeFromFilter(dateFilter, BEFORE);
+            if (before is string) {
+                queryParams += "&" + convertToUnderscoreCase(key) + SUFFIX_MAX + "=" + before;
+            }
+            string? after = getTimeStringTimeFromFilter(dateFilter, AFTER);
+            if (after is string) {
+                queryParams += "&" + convertToUnderscoreCase(key) + SUFFIX_MIN + "=" + after;
+            }
+        }
+    }
+    if (queryParams == "") {
+        return queryParams;
+    } else {
+        // Remove starting '&' character from the query parameters
+        return "?" + 'string:substring(queryParams, 1, queryParams.length());
+    }
+}
+
+function notImplemented() returns Error {
```

Comment:
Do we need this function?
Since we are doing only returning as an error. I think we can remove it